### ISSUE TITLE
Code change causes warning with armcc compiler

### ISSUE THREAD

**Issue by CurtisAMU** [ETLCPP/etl#749]
https://github.com/ETLCPP/etl/blob/b38afd7fc98f50a00b207b184506049f05ab7120/include/etl/private/vector_base.h#L171

The extra semicolon causes a warning if `ETL_DEBUG_COUNT` is not defined:
```
..\..\..\Lib\ETL\include\etl/private/vector_base.h(171): warning:  #381-D: extra ";" ignored
      ETL_DECLARE_DEBUG_COUNT;   ///< Internal debugging.
```

We have a zero warning policy for our project, which treats this as an error. It would be okay for debugging to have a warning to ignore but preferably not on release builds.

**Comment by jwellbelove** [comment]
Does it warn with `void;`

**Comment by jwellbelove** [comment]
Forget that last suggestion

**Comment by jwellbelove** [comment]
What about `void(0);`

**Comment by CurtisAMU** [comment]
`ETL_DECLARE_DEBUG_COUNT;` is a declaration and `(void)0;` does not work there; it's an expression.
The former
`ETL_DECLARE_DEBUG_COUNT   ///< Internal debugging.`
without the semicolon worked. What was the problem with that? It looks odd without the semicolon but should work.

This also works with armcc:
`#define ETL_DECLARE_DEBUG_COUNT struct SDebugCountDummy {}`

**Comment by CurtisAMU** [comment]
A dummy typedef would also do.

**Comment by jwellbelove** [comment]
There will also be this warning if `ETL_NO_CHECKS` is defined for `ETL_ASSERT` etc.

**Comment by jwellbelove** [comment]
Do the other macros cause a warning?
`ETL_INCREMENT_DEBUG_COUNT` etc

**Comment by CurtisAMU** [comment]
Actually no. Empty statements are okay, it seems.

**Comment by CurtisAMU** [comment]
> Do the other macros cause a warning? `ETL_INCREMENT_DEBUG_COUNT` etc

Not as far as I can see.

**Comment by CurtisAMU** [comment]
They are not used in a class declaration.

**Comment by jwellbelove** [comment]
If I try an empty statement in a Keil project using ARM compiler V6 I get

`warning: empty expression statement has no effect; remove unnecessary ';' to silence this warning [-Wextra-semi-stmt]`

**Comment by CurtisAMU** [comment]
One of those complementary differences between compiler V5 and V6. There `(void)0;` would help in both cases, I'd expect.

**Comment by jwellbelove** [comment]
Ignore that file, there's an error.

**Comment by CurtisAMU** [comment]
Looks good, the warning is gone.

**Comment by jwellbelove** [comment]
Fixed 20.38.1
